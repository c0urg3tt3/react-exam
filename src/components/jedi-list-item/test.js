import React from 'react'

import { shallow } from 'enzyme'

import JediListItem from './component'

describe('[component] JediListItem', () => {
  it('should render component', () => {
    const props = {
      jedi: { id: 1337, name: 'Anakin Skywalker' }
    }

    const wrapper = shallow(<JediListItem/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('li')).toEqual(true)
    expect(wrapper.hasClass('jedi-list-item')).toEqual(true)

    let span = wrapper.find('span')
    expect(span).toHaveLength(2)
    expect(span.first().text()).toBe("")
    expect(span.last().text()).toBe("")
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps(props)

    span = wrapper.find('span')
    expect(span).toHaveLength(2)
    expect(span.first().text()).toBe(props.jedi.id +"")
    expect(span.last().text()).toBe(props.jedi.name)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
