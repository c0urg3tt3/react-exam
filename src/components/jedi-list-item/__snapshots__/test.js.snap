// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediListItem should render component 1`] = `"<li class=\\"jedi-list-item\\"><span></span><span></span></li>"`;

exports[`[component] JediListItem should render component 2`] = `"<li class=\\"jedi-list-item\\"><span>1337</span><span>Anakin Skywalker</span></li>"`;
